### DISCUSSION TITLE
Unindenting content?

### DISCUSSION THREAD

**Discussion by TrisMcC** [micromark/micromark#96] [Q&A] [answered]
Hello,

I am trying to implement https://facelessuser.github.io/pymdown-extensions/extensions/tabbed/ in micromark to use in a web application. I've gone back and forth as to whether to implement this at the remark level or the micromark level. As of now, I am doing it with micromark. I am using `text` as the extension key with `equalsTo` as the code that kicks the whole thing off.

I am unsure how to "unindent" the content that is in the tabs. I am capturing all codes as chunkContent in the tab content area and I cannot seem to capture the spaces as something else just to "throw them away". When I naively try to do that, I get "expected last token to be open".

Is what I am asking for possible? I searched for an existing implementation of this tab type but came up empty. Is there a similar implementation anywhere?

Thanks!

gist: https://gist.github.com/TrisMcC/0b37c96d2eb34e4c2736c1b5b30c54e7

**Comment by wooorm** [comment, marked as answer]
- almost everything is possible!
- should you do it? Probably no. See the stuff on extensions in the readme
- if you‘re going to do it, is it hard? yes, yes it is.
- everything has to be a token. Unknown tokens are thrown away.
- you can’t just eat every character other than a space and put it into one token! At least line endings, but several things like EOFs need to be handled!
- Why not use directives?!
